mava/systems/tf/variable_utils.py

```diff
@@ -0,0 +1,154 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Variable handling utilities for TensorFlow 2. Adapted from Deepmind's Acme library"""
+
+from concurrent import futures
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import tensorflow as tf
+from acme.tf import utils as tf2_utils
+
+from mava.systems.tf.variable_sources import VariableSource as MavaVariableSource
+
+
+class VariableClient:
```

Comment:
What changed here compared to the acme client?

Reply:
I made this client more general. The Acme client could just pull variables every few steps. This variable client provides functions to set and get variables from the centralised variable source.